Bluetooth: controller: split: fixes DLE conformance tests

kruithofa opened the pull request:
For review by Vinayak and EBQ verification by Morten
EBQ run with CODED and 2M phy configured in without failures

Fixes #22048 except for LL/CON/SLA/BV-131-C
See https://github.com/EDTTool/EDTT/pull/12 for a fix for the EDTT

Commit:
Bluetooth: controller: split: fix LL/CON/MAS/BV-74-C tx timing

Fix EBQ tests for the Max Tx Time and Max Rx Time parameter.

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

Bluetooth: controller: split: fixed for endianness

Added conversion to correct endianness

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -3191,38 +3243,13 @@ static inline void event_len_prep(struct ll_conn *conn)
 		lr->max_tx_time = sys_cpu_to_le16(tx_time);
 
-
+#if defined(CONFIG_BT_CTLR_PHY)
```
revert this change, so that `conn->default_tx_time` is not changed and retry

kruithofa replied:
Change implemented

cvinayak commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -4494,35 +4521,30 @@ static inline int length_req_rsp_recv(struct ll_conn *conn, memq_link_t *link,
-		if (max_rx_time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
-			eff_tx_time = MIN(max_rx_time,
-					  conn->default_tx_time);
+		if (lr->max_rx_time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
```
put back the `sys_le16_to_cpu`

mtpr-ot replied:
Yes, these conversions must be applied to references to lr->max_xxxxx. Should probably be done by assigning a local variable

kruithofa replied:
introduced local vars

mtpr-ot commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -4493,14 +4513,17 @@ static inline int length_req_rsp_recv(struct ll_conn *conn, memq_link_t *link,
+
+		dle_max_time_get(conn, &temp_rx, &temp_tx);
+		max_rx_time = sys_le16_to_cpu(temp_rx);
+		max_tx_time = sys_le16_to_cpu(temp_tx);
```
This looks incorrect. Why would you endianness-convert from LE, when value is provided by dle_max_time_get?

kruithofa replied:
as per suggestion from Vinayak; we were seeing some issues trying to resolve

kruithofa replied:
removed

mtpr-ot commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -3127,6 +3127,76 @@ static inline void event_ping_prep(struct ll_conn *conn)
+	u32_t feature_coded_phy;
+	u32_t feature_phy_2m;
+	u16_t rx_time;
+	u16_t tx_time;
```
Both need to be initialized to zero, to avoid compiler warning

kruithofa replied:
initialization added

mtpr-ot commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -4511,18 +4534,12 @@ static inline int length_req_rsp_recv(struct ll_conn *conn, memq_link_t *link,
 		 */
-		max_tx_time = sys_le16_to_cpu(lr->max_tx_time);
-		if (max_tx_time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
+		if (lr->max_tx_time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
```
sys_le16_to_cpu

mtpr-ot commented on subsys/bluetooth/controller/ll_sw/ull_conn.c:
```diff
@@ -4511,18 +4534,12 @@ static inline int length_req_rsp_recv(struct ll_conn *conn, memq_link_t *link,
-		max_tx_time = sys_le16_to_cpu(lr->max_tx_time);
-		if (max_tx_time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
+		if (lr->max_tx_time >= PKT_US(PDU_DC_PAYLOAD_SIZE_MIN, 0)) {
+			eff_rx_time = MIN(lr->max_tx_time, max_rx_time);
```
sys_le16_to_cpu x 2 (or local var)

kruithofa replied:
Introduced local variable

mtpr-ot reviewed (changes requested):
Please update endianness handling as suggested.

kruithofa commented:
Shippable fails, requires an update to EDTT

mtpr-ot reviewed (approved):
LGTM. EBQ tests pass at our end.

jhedberg reviewed (approved):
Looks fine, but we need to get Shippable passing before merging

aescolar commented:
@kruithofa Please note that this PR seems to be quite behind master, 
I did a quick trial rebasing on top of master and referencing in the manifest to the EDTT PR, but a few tests seem to fail(?). You can see it here:
https://github.com/aescolar/zephyr/tree/alpi_kruithofa

Commit:
manifest: EDTT: update manifest to point to new revision in EDTT

For passing the EDTT tests pass/fail criteria needs to be updated,
which is done starting with the revision referenced in the updated
manifest

Signed-off-by: Andries Kruithof <Andries.Kruithof@nordicsemi.no>

aescolar reviewed (dismissed):
Just to avoid accidental merges: Careful that it points to a west module PR
(anybody, feel free to dismiss whenever it does not anymore)
